src/browser/existing-browser.ts

```diff
@@ -111,6 +118,67 @@ export class ExistingBrowser extends Browser {
         return this;
     }
 
+    async startCollectCookies(): Promise<void> {
+        if (!this._session) {
+            return;
+        }
+
+        this.allCookies = new Map();
+
+        this.publicAPI.addCommand("getAllRequestsCookies", async () => {
```

Comment:
Here, we add a whole new command to user-facing API. Is this intentional?

How is it different from `getAllCookies`? The names are very similar and it's confusing to have them both.

I see we are using it in saveState. Perhaps we should make it a method of existing browser class that's unavailable to users instead of a new command?